### PR TITLE
Add mock interfaces

### PR DESCRIPTION
**Pull request by justaugustus**
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

**What this PR does / why we need it**:
This is a first pass at adding mocks to the project. 
I've deleted the old (pre-refactor) mock interfaces and leveraged the gomock and the [bazel generation rules](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/build/go_mock.bzl) to create new mocks.

With new mocks, we can start to deepen our test coverage across the Azure resource-level packages.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
This is needed to enable some testing work on #174.

**Release note**:

```release-note
Add mock interfaces
```

### CODE REVIEW

File: pkg/cloud/azure/actuators/cluster/reconciler.go [hunk 1]
```diff
@@ -35,15 +36,15 @@ import (
...
-	certificatesSvc  azure.Service
-	groupsSvc        azure.Service
-	vnetSvc          azure.Service
-	securityGroupSvc azure.Service
-	routeTableSvc    azure.Service
-	subnetsSvc       azure.Service
-	internalLBSvc    azure.Service
-	publicIPSvc      azure.Service
-	publicLBSvc      azure.Service
+	certificatesSvc  services.Service
```
Comment (awesomenix) [inline pkg/cloud/azure/actuators/cluster/reconciler.go]: please change it back to azure.Service, since services.Service has a stutter.

Comment (awesomenix) [reply pkg/cloud/azure/actuators/cluster/reconciler.go]: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/138

**Comment by justaugustus** [comment]
(Pivoting a little on the mocks to utilize [bazel_gomock](https://github.com/jmhodges/bazel_gomock), so I'm setting this as WIP again.)

File: pkg/cloud/azure/actuators/cluster/reconciler.go (hunk 1 above)
Comment (justaugustus) [reply pkg/cloud/azure/actuators/cluster/reconciler.go]: Reverted that change.

**Comment by justaugustus** [comment]
Cross-linking https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/719 since our mock [tests](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/182/pull-cluster-api-provider-azure-bazel-test/1118095581219131392) are now failing like @vincepri's [tests](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/719/pull-cluster-api-provider-aws-bazel-test/1117971140313092097)

**Comment by justaugustus** [comment]
(Re-running tests for paranoia's sake)
/retest

**Comment by justaugustus** [comment]
Builds were failing due to the images we are using for this repo's presubmits. After merging https://github.com/kubernetes/test-infra/pull/12217, all is well!

**Pull request merged**